### PR TITLE
Harmonize interface of methods and create unified interface

### PR DESCRIPTION
**Pull request by MaxBlesch**

### CODE REVIEW

File: src/upper_envelope/fues_numba/fues_numba.py [hunk 1]
```diff
@@ -70,7 +73,6 @@ def fast_upper_envelope_wrapper(
             containing refined state- and choice-specific value function.
 
     """
-    n_grid_wealth = len(endog_grid)
     min_wealth_grid = np.min(endog_grid)
     # exog_grid = np.append(
```
Comment (segsell) [inline src/upper_envelope/fues_numba/fues_numba.py]: Can we remove the commented out part?

File: src/upper_envelope/fues_numba/fues_numba.py [hunk 2]
```diff
@@ -163,11 +178,11 @@ def fast_upper_envelope(
...
-    mask = endog_grid <= lower_bound_wealth
-    if np.any(mask):
-        max_value_lower_bound = np.nanmax(value[mask])
-        mask &= value < max_value_lower_bound
-        value[mask] = np.nan
+    # mask = endog_grid <= tuning_params["lower_bound_wealth"]
+    # if np.any(mask):
+    #     max_value_lower_bound = np.nanmax(value[mask])
+    #     mask &= value < max_value_lower_bound
+    #     value[mask] = np.nan
```
Comment (segsell) [inline src/upper_envelope/fues_numba/fues_numba.py]: Do we need the commented-out part still?

File: src/upper_envelope/fues_jax/fues_jax.py [hunk 3]
```diff
@@ -182,6 +201,11 @@ def fast_upper_envelope(
     #     mask &= value < max_value_lower_bound
     #     value[mask] = jnp.nan
```
Comment (segsell) [inline src/upper_envelope/fues_jax/fues_jax.py]: Do we need the commented-out part still?

File: src/upper_envelope/fues_numba/fues_numba.py [hunk 4]
```diff
@@ -163,11 +178,11 @@ def fast_upper_envelope(
     """
     # TODO: determine locations where endogenous grid points are # noqa: T000
```
Comment (segsell) [inline src/upper_envelope/fues_numba/fues_numba.py]: Do we need the TODO note still?

**Review by segsell — COMMENTED** [review]
Very nice PR! I really like the handy jit implementations. See some minor comments below.

File: src/upper_envelope/fues_jax/fues_jax.py [hunk 5]
```diff
@@ -80,35 +94,35 @@ def fast_upper_envelope_wrapper(
...
-        utility_function,
-        utility_kwargs,
-        disc_factor,
+    values_to_add = vmap(_compute_value, in_axes=(0, None, None, None))(
+        grid_points_to_add, value_function, value_function_args, value_function_kwargs
     )
 
-    # Now determine if we actually had to extend the grid. If not, we just add nans.
+    # Now determine if n_constrained_points_to_addwe actually had to extend the grid.
+    # If not, we just add nans.
```
Comment (segsell) [inline src/upper_envelope/fues_jax/fues_jax.py:137]: This comment seems a little confusing. Can you rewrite it?

File: tests/utils/interpolation.py [hunk 6]
```diff
@@ -157,10 +157,10 @@ def linear_interpolation_with_extrapolation(x, y, x_new):
     """Linear interpolation with extrapolation.
 
     Args:
-        x (np.ndarray): 1d array of shape (n,) containing the x-values.
-        y (np.ndarray): 1d array of shape (n,) containing the y-values
+        x : 1d array of shape (n,) containing the x-values.
+        y : 1d array of shape (n,) containing the y-values
             corresponding to the x-values.
-        x_new (np.ndarray or float): 1d array of shape (m,) or float containing
+        x_new: 1d array of shape (m,) or float containing
```
Comment (segsell) [inline tests/utils/interpolation.py]: Can you add the input types back to the docstrings? Is it ```np.ndarray```, ```jnp.ndarray```, and ```float```?

File: src/upper_envelope/fues_numba/fues_numba.py (hunk 1 above)
Comment (MaxBlesch) [reply src/upper_envelope/fues_numba/fues_numba.py]: Removed!

File: src/upper_envelope/fues_numba/fues_numba.py (hunk 2 above)
Comment (MaxBlesch) [reply src/upper_envelope/fues_numba/fues_numba.py]: Removed. Is mentioned in Issue and we need to discuss with Akshay why he had that.

File: src/upper_envelope/fues_numba/fues_numba.py (hunk 4 above)
Comment (MaxBlesch) [reply src/upper_envelope/fues_numba/fues_numba.py]: Removed

File: tests/utils/interpolation.py (hunk 6 above)
Comment (MaxBlesch) [reply tests/utils/interpolation.py]: POlished!

File: src/upper_envelope/fues_jax/fues_jax.py (hunk 3 above)
Comment (MaxBlesch) [reply src/upper_envelope/fues_jax/fues_jax.py]: Removed

File: src/upper_envelope/fues_jax/fues_jax.py (hunk 5 above)
Comment (MaxBlesch) [reply src/upper_envelope/fues_jax/fues_jax.py:137]: Rewrote. Please check if it is understandable now!

Comment (MaxBlesch) [reply src/upper_envelope/fues_jax/fues_jax.py:137]: Just a typo. Done now,

**Review by segsell — APPROVED** [review]
Looking great! Thank you 🎉

**Pull request merged**